### ISSUE TITLE
Add CSV export function to Hive NYC directory 

### ISSUE THREAD

**Issue by jazoulai** [mozilla/learning.mozilla.org#1732]
Can someone please help us add a CSV export function to the [Hive NYC directory](https://directory.hivelearningnetworks.org/nyc/)?

Adding @jvallera so she can watch this issue as well.

Thanks!

**Comment by hannahkane** [comment]
hey @jazoulai - thanks for filing. Can you provide a little more information to help us prioritize? Do you need this by a certain date? If we were able to do a one-time export and send to you, would that be adequate for now, or do you need the function on an ongoing basis?

**Comment by jazoulai** [comment]
Thanks for your quick response @hannahkane. I think for now a one-time export would work for our purposes. For future needs it would be great to have the CSV export feature so that we don't have to make a request each time we need an updated CSV.

In case the larger context is useful to you, we are creating an interactive map using [CartoDB](https://cartodb.com/) linked to a Google spreadsheet. My plan is to pull the directory information as a CSV, add GeoData to each row, and visualize the data on a map. We'd like to update the map data on an ongoing basis,  as new organizations sign up on the directory or current organizations edit their info.

Ideally, it would be great to have access via an API that can feed into the Google spreadsheet, but I imagine that would require way more work than a CSV export feature. Definitely not trying to take up much of your team's bandwidth for this! Thanks so much :)

**Comment by hannahkane** [comment]
Thanks, @jazoulai. I'm putting this in the next heartbeat (April 8), because the current heartbeat is pretty packed. We looked into this a little on Friday, and since it may not be completely trivial, I'd rather not have the team focus on it until the beginning of the next week. Let me know if there's an external deadline that would impact this.

**Comment by jazoulai** [comment]
Works for us, @hannahkane. Thanks!

**Comment by jvallera** [comment]
I have another Hive NYC directory related issue so adding it here, but happy to start a new ticket if that is easier. I would like to add Joey and Hana as administrators so they can have full access drupal dashboard, but I don't have the ability to do that. Any thoughts on how we can do this @hannahkane ? Right now Lainie, Chris, Atul, Leah and I are the people with access to the dashboard.

**Comment by hannahkane** [comment]
cc @simonwex to answer @jvallera's question.

**Comment by simonwex** [comment]
@jvallera we're in a funny situation here, as I don't have an account. We've created a devops bug to track this here: https://github.com/MozillaFoundation/mofo-devops/issues/299

**Comment by jvallera** [comment]
Thanks @simonwex Will add myself to #299 @jazoulai can be copied as well. Let us know if there anything we can do. Happy to give you my login if it helps.

**Comment by mmmavis** [comment]
hey @jazoulai what info would you like to see in the exported .csv? name and city of organizations?

**Comment by jazoulai** [comment]
Hi @mmmavis. Would be great to have all user-facing fields, but no need for any metadata. Here' s a list based on what I can see on the front-end of the directory, as far as organizations:

1. Name
2. Website
3. Address
4. Twitter name
5. Mission
6. Content Channels

Also, I think I could make use of user profile data as well, if including that in the export doesn't require much extra work. But I can hold off otherwise.

1. username
2. First name
3. Last name
4. Title
5. Bio
6. Twitter name
7. Expertise

**Comment by mmmavis** [comment]
Hey @jazoulai ,

Can I propose that we have the CSV export functionality to export just the organizations' `name`, `website`, `address`, `twitter name`, and `mission` for now? It seems like "content channel" lives as a separate Django model (e.g., it's not part of the Organization model). I spent quite a while trying to export content channel info but still couldn't get it working. As a Python/Django super novice I can't estimate how long it will take me to have the exported CSV include "content channel" - could be a few more days. :fearful: 

If you are okay with leaving "content channel" out, I should be able to get the export feature out soon. I have it working on my machine - just need to figure out how to deploy the changes live.

(Exporting user profiles seems to be more complicate to implement. I would suggest that we leave it out for now.)

**Comment by jazoulai** [comment]
Sounds good @mmmavis, thank you! 

Are we talking about a one-time export of the data, or export functionality?

**Comment by jvallera** [comment]
thank you @mmmavis !! you are working wonders. I thought we would never be able to get this info ;) Thanks @jazoulai for working on this.

**Comment by mmmavis** [comment]
@jazoulai , same as @simonwex , I don't have an account on the production site Admin portal/database... which means I don't have access to the "real data".  I can help add an Export to CSV functionality to the dashboard though, since I have pushing rights to the Github repo and Heroku.

@jvallera :relaxed: glad i can help!

**Comment by mmmavis** [comment]
encountered various problems when I tried to deploy changes live. Will provide updates once the problems are resolved.

**Comment by mmmavis** [comment]
@jazoulai I pushed the updates to production just now :tada: Let me know if the CSV export functionality works as expected :grin:

**Comment by jazoulai** [comment]
Yay!!! So exciting, thanks @mmmavis! Sorry it gave you such a headache.

I don't think I have access to the admin dash myself, @jvallera will you take a look and report back to us?

**Comment by jvallera** [comment]
Thanks @mmmavis I don't see anything in the dashboard to export. Do you know where I should be looking? or how I access it?

**Comment by mmmavis** [comment]
@jvallera you should see a **Export CSV** option in the dropdown when you are on the  "Organizations" admin page.

![screen shot 2016-04-11 at 1 40 00 pm](https://cloud.githubusercontent.com/assets/2896608/14441702/e967cfda-ffea-11e5-8626-3a5b2a258aee.png)

**Comment by jvallera** [comment]
Hmm, don't see it on my end @mmmavis 
![screen shot 2016-04-11 at 4 47 47 pm](https://cloud.githubusercontent.com/assets/3729312/14441948/4b9d41de-0005-11e6-8aba-d054daf0b1a5.png)

**Comment by mmmavis** [comment]
:sob:
@jvallera could you check again? I'm hoping it was some caching / delay issue.

**Comment by jvallera** [comment]
still no luck @mmmavis tried through a different browser too. 
![screen shot 2016-04-20 at 1 16 50 pm](https://cloud.githubusercontent.com/assets/3729312/14683777/7b6c58c2-06fa-11e6-8e88-31db6e21cb5c.png)

**Comment by hannahkane** [comment]
@mmmavis - do you have an update on this ticket?

**Comment by simonwex** [comment]
I think this is blocked on https://github.com/MozillaFoundation/mofo-devops/issues/299 which should be resolved today.

**Comment by hannahkane** [comment]
Looks like this is still blocked, so moving to an upcoming heartbeat to keep tracking it.
